docs/dockerfile_experimental.md

```diff
@@ -0,0 +1,128 @@
+|`id`                 | ID of the cache.|
+|`target`(required)   | Mount path.|
+|`ro`,`readonly`      | Read-only if set.|
+|`sharing`            | Either one of `shared`, `private`, or `locked`. Defaults to `shared`. `shared` cache mount can be used concurrently by multiple writers. `private` creates a new mount if there are multiple writers. `locked` pauses second writer until first one releases the mount.|
+
+
+Example: cache Go packages
+
+```dockerfile
+# syntax = tonistiigi/dockerfile:master-experimental
+FROM golang
+...
+RUN --mount=type=cache,target=/root/.cache/go-build go build ...
+```
+
+Example: cache apt packages
+
+```dockerfile
+# syntax = tonistiigi/dockerfile:master-experimental
+FROM ubuntu
+RUN rm -f /etc/apt/apt.conf.d/docker-clean; echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache
+RUN --mount=type=cache,target=/var/cache/apt --mount=type=cache,target=/var/lib/apt \
+  apt update && apt install -y gcc
+```
+
+### `RUN --mount=type=tmpfs`
+
+This mount type allows mounting tmpfs in the build container.
+
+|Option               |Description|
+|---------------------|-----------|
+|`target`(required)   | Mount path.|
+
+
+### `RUN --mount=type=secret`
+
+This mount type allows the build container to access credential files such as AWS keys without baking them into the image.
+
+|Option               |Description|
+|---------------------|-----------|
+|`id`                 | ID of the secret. Defaults to `path.Base(target)`.|
+|`target`             | Mount path. Defaults to `/run/secrets/ + path.Base(id)`.|
+|`required`           | If set to `true`, the instruction errors out when the secret is unavailable. Defaults to `false`.|
+
+
+Example: access to S3
+
+```dockerfile
+# syntax = tonistiigi/dockerfile:master-experimental
+FROM python:3
+RUN pip install awscli
+RUN --mount=type=secret,id=aws,target=/root/.aws/credentials aws s3 cp s3://... ...
+```
+
+```console
+$ buildctl build --frontend=dockerfile.v0 --local context=. --local dockerfile=. \
+  --secret id=aws,src=$HOME/.aws/credentials
+```
+
+### `RUN --mount=type=ssh`
+
+This mount type allows the build container to access SSH keys, with support for passphrases.
+
+|Option               |Description|
+|---------------------|-----------|
+|`required`           | If set to `true`, the instruction errors out when the key is unavailable. Defaults to `false`.|
+
+
+Example: access to Gitlab
+
+```dockerfile
+# syntax = tonistiigi/dockerfile:master-experimental
+FROM alpine
+RUN apk add --no-cache openssh-client
+RUN mkdir -p -m 0700 ~/.ssh && ssh-keyscan gitlab.com >> ~/.ssh/known_hosts
+RUN --mount=type=ssh ssh git@gitlab.com | tee /hello
+# "Welcome to GitLab, @GITLAB_USERNAME_ASSOCIATED_WITH_SSHKEY" should be printed here
+```
+
+```console
+$ eval $(ssh-agent)
+$ ssh-add ~/.ssh/id_rsa
+(Input your passphrase here)
+$ buildctl build --frontend=dockerfile.v0 --local context=. --local dockerfile=. \
```

Comment:
Add an example for using the keys directly (with a filename `myproject_deploy_key.pem`)